python/GafferSceneUI/SceneViewUI.py

```diff
@@ -1073,13 +1076,105 @@ def __appendClippingPlaneMenuItems( menuDefinition, prefix, view, parentWidget )
 			}
 		)
 
+def __snapshotDescription( view ) :
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	if sceneGadget.getRenderer() == "OpenGL" :
+		return "Viewport snapshots are only available for rendered (non-OpenGL) previews."
+
+	return "Snapshot viewport and send to catalogue."
+
+def __snapshotToCatalogue( catalogue, view ) :
+
+	timeStamp = str( datetime.datetime.now() )
+
+	fileName = pathlib.Path(
+		view["in"].node().getContext().substitute( catalogue["directory"].getValue() )
+	) / ( timeStamp.replace( ":", "-" ).replace(" ", "_" ) + ".exr" )
+
+	resolutionGate = imath.Box3f()
+	if isinstance( view, GafferSceneUI.SceneView ) :
+		resolutionGate = view.resolutionGate()
+
+	scriptRoot = view["in"].getInput().ancestor( Gaffer.ScriptNode )
+	metadata = IECore.CompoundData(
+		{
+			"gaffer:sourceScene" : view["in"].getInput().relativeName( scriptRoot ),
+			"gaffer:context:frame" : view["in"].node().getContext().getFrame()
+		}
+	)
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	sceneGadget.snapshotToFile( fileName, resolutionGate, metadata )
+
+	image = GafferImage.Catalogue.Image( "Snapshot1", Gaffer.Plug.Direction.In, Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+	image["fileName"].setValue( fileName )
+
+	image["description"].setValue(
+		"Snapshot of {} at frame {} taken at {}".format(
+			metadata["gaffer:sourceScene"],
+			metadata["gaffer:context:frame"], timeStamp[:-6]  # Remove trailing microseconds
+		)
+	)
+
+	catalogue["images"].source().addChild( image )
```

Comment:
Somewhere here I need to set the current image in the catalog to be the one just added